doc/setup_instructions.md

```diff
@@ -558,6 +563,19 @@ files to check.
 
 ## Additional Tips
 
+### Troubleshooting regarding Godot automatically breaking
+
+Godot sometimes likes to break your files for no reason. If you keep
+the Godot editor open while pulling new changes or changing branches,
+it's very likely to break so it is recommended to close Godot when
+doing such operations that change files outside the Godot editor, and
+then reopening the editor afterwards.
```

Comment:
That would be nice, but the issue about adding more explicitly text talking about this has been open for a long time. If in the future Godot actually works properly here, this text might be suitable to be deleted, but for now I want to add this because I'm tired of explaining this to people.